Sources/MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -695,6 +696,42 @@ open class CarPlayNavigationViewController: UIViewController, BuildingHighlighti
         navigationMapView?.showWaypoints(on: progress.route, legIndex: legIndex)
     }
     
+    func updateLegacyIcons() {
+        spriteRepository.legacyCache.clearMemory()
+        updateManeuvers(navigationService.routeProgress)
+    }
+    
+    func updateInstruction(_ routeProgress: RouteProgress) {
+        guard spriteRepository.getSpriteImage() == nil else {
+            updateManeuvers(routeProgress)
+            return
+        }
+        
+        spriteRepository.updateSprite(styleURI: spriteRepository.styleURI) { [weak self] in
+             guard let self = self else { return }
+             self.updateManeuvers(self.navigationService.routeProgress)
```

Comment:
After downloading the Sprite, using the current `routeProgress` for Maneuver labeling, in case of the delay.